packages/context/src/value-promise.ts

```diff
@@ -50,7 +50,10 @@ export function isPromiseLike<T>(
  * @param value Value of an object
  * @param path Path to the property
  */
-export function getDeepProperty(value: BoundValue, path: string): BoundValue {
+export function getDeepProperty<T = BoundValue>(
```

Comment:
While you are changing the signature of this function, would it make sense to give `value` argument a generic type too?

```ts
export function getDeepProperty<V, T = BoundValue>(
  value: V,
  path: string
): T {
  // impl
}
```

Ideally, I'd like to get rid of as many usages of `any` type as we can, eventually removing `BoundValue` alias too.

Feel free to leave such change out of this pull request if you disagree or if it's not trivial to make.

Reply:
I have struggled between option A and B:

A:
```ts
export function getDeepProperty<V = BoundValue, T = BoundValue>(
  value: V,
  path: string,
): T | undefined {// ...}
```

B:
```ts
export function getDeepProperty<T = BoundValue, V = BoundValue>(
  value: V,
  path: string,
): T | undefined {// ...}
```

For A, type `V` and `T` follows the convention as parameter and return types. But `T` is more important than `V`. With A, we cannot use `getDeepProperty<string>` anymore if the result is expected to `string`. Instead, we have to use `getDeepProperty<V, string>`. It's a bit awkward to me.

For B, `T` comes before `V` and it's not so natural either.

Anyhow, getDeepProperty is mostly used internally. So I go with `A` for now.